app/src/main/java/fi/aalto/legroup/achso/entities/OptimizedVideo.java

```diff
@@ -0,0 +1,359 @@
+        // annotations and the location object if needed.
+        Video video = pooled.create(annotationCount, hasLocation);
+
+        // Parse the Uri objects back from strings
+        video.setManifestUri(Uri.parse(manifestUri));
+        video.setVideoUri(Uri.parse(videoUri));
+        video.setThumbUri(Uri.parse(thumbUri));
+
+        video.setRepository(repository);
+        video.setId(id);
+        video.setTitle(title);
+        video.setGenre(genre);
+        video.setTag(tag);
+
+        // Create the Date objects from the longs
+        video.setDate(new Date(dateInMs));
+        if (hasLastModified) {
+            video.setLastModified(new Date(lastModifiedInMs));
+        } else {
+            video.setLastModified(null);
+        }
+
+        // Retrieve the author with the index from the user pool
+        video.setAuthor(UserPool.getInternedUser(authorUserIndex));
+
+        // Recreate the location from the location fields.
+        // Note: If we are creating the video in place there might be a Location object already,
+        // so we can just fill the data in there instead of creating a new Location object.
+        if (hasLocation) {
+            Location location = video.getLocation();
+            if (location == null) {
+                location = new Location("inflated");
+                video.setLocation(location);
+            }
+            location.setLatitude(locationLatitude);
+            location.setLongitude(locationLongitude);
+            location.setAccuracy(locationAccuracy);
+        } else {
+            video.setLocation(null);
+        }
+
+        List<Annotation> annotations = video.getAnnotations();
+        for (int i = 0; i < annotationCount; i++) {
+
+            // The video comes from PooledVideo it has already allocated the list of the
+            // annotations, see beginning of this function
+            Annotation annotation = annotations.get(i);
+
+            annotation.setTime(annotationTime[i]);
+
+            // The location object should exist for the annotation already.
+            PointF position = annotation.getPosition();
+            if (position == null) {
+                position = new PointF();
+                annotation.setPosition(position);
+            }
+            position.set(annotationXY[i * 2], annotationXY[i * 2 + 1]);
+
+            // Retrieve the substring from the annotation text buffer.
+            annotation.setText(getAnnotationText(i));
+
+            // Retrieve the interned user with the index.
+            annotation.setAuthor(UserPool.getInternedUser(annotationAuthorUserIndex[i]));
+        }
+
+        return video;
+    }
+
+    /**
+     * Inflate the optimized video data into a heavier Video object.
+     * Note: This version allocates a new video object, you probably should use some static
+     * PooledVideo and use `inflate` instead of this.
+     */
+    public Video inflateNew() {
```

Comment:
In my opinion it’s not that difficult to figure out that if it’s not given a `PooledVideo`, it will instantiate one itself.

The same method name would imply that it’s meant for accomplishing the same end result, but through a different course of action (i.e. doing something extra.)